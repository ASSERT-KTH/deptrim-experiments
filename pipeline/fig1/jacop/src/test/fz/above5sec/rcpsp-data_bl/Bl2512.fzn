predicate jacop_cumulative(array [int] of var int: s,array [int] of var int: d,array [int] of var int: r,var int: b);
array [1..2] of int: X_INTRODUCED_27 = [1,-1];
array [1..15] of int: X_INTRODUCED_46 = [1,2,2,2,1,1,2,1,2,2,1,2,2,2,1];
array [1..15] of int: X_INTRODUCED_47 = [2,4,5,5,3,4,3,5,3,4,3,1,2,5,4];
array [1..15] of int: X_INTRODUCED_49 = [2,2,1,1,2,1,1,2,1,1,2,2,2,2,1];
array [1..15] of int: X_INTRODUCED_50 = [4,5,5,5,5,3,3,5,4,3,2,5,2,5,5];
array [1..22] of int: X_INTRODUCED_52 = [3,3,3,3,1,3,1,3,1,3,3,3,2,1,2,1,2,3,1,3,2,3];
array [1..22] of int: X_INTRODUCED_53 = [3,2,2,4,5,5,5,5,3,4,3,5,1,4,1,2,5,2,5,3,5,4];
var 0..86: X_INTRODUCED_0;
var 0..86: X_INTRODUCED_1;
var 0..86: X_INTRODUCED_2;
var 0..86: X_INTRODUCED_3;
var 0..86: X_INTRODUCED_4;
var 0..86: X_INTRODUCED_5;
var 0..86: X_INTRODUCED_6;
var 0..86: X_INTRODUCED_7;
var 0..86: X_INTRODUCED_8;
var 0..86: X_INTRODUCED_9;
var 0..86: X_INTRODUCED_10;
var 0..86: X_INTRODUCED_11;
var 0..86: X_INTRODUCED_12;
var 0..86: X_INTRODUCED_13;
var 0..86: X_INTRODUCED_14;
var 0..86: X_INTRODUCED_15;
var 0..86: X_INTRODUCED_16;
var 0..86: X_INTRODUCED_17;
var 0..86: X_INTRODUCED_18;
var 0..86: X_INTRODUCED_19;
var 0..86: X_INTRODUCED_20;
var 0..86: X_INTRODUCED_21;
var 0..86: X_INTRODUCED_22;
var 0..86: X_INTRODUCED_23;
var 0..86: X_INTRODUCED_24;
var 0..87: objective:: output_var;
array [1..25] of var int: s:: output_array([1..25]) = [X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_8,X_INTRODUCED_9,X_INTRODUCED_10,X_INTRODUCED_11,X_INTRODUCED_12,X_INTRODUCED_13,X_INTRODUCED_14,X_INTRODUCED_15,X_INTRODUCED_16,X_INTRODUCED_17,X_INTRODUCED_18,X_INTRODUCED_19,X_INTRODUCED_20,X_INTRODUCED_21,X_INTRODUCED_22,X_INTRODUCED_23,X_INTRODUCED_24];
array [1..15] of var int: X_INTRODUCED_48 ::var_is_introduced  = [X_INTRODUCED_1,X_INTRODUCED_3,X_INTRODUCED_5,X_INTRODUCED_7,X_INTRODUCED_8,X_INTRODUCED_10,X_INTRODUCED_11,X_INTRODUCED_12,X_INTRODUCED_14,X_INTRODUCED_15,X_INTRODUCED_16,X_INTRODUCED_17,X_INTRODUCED_20,X_INTRODUCED_23,X_INTRODUCED_24];
array [1..15] of var int: X_INTRODUCED_51 ::var_is_introduced  = [X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_8,X_INTRODUCED_11,X_INTRODUCED_12,X_INTRODUCED_15,X_INTRODUCED_16,X_INTRODUCED_18,X_INTRODUCED_19,X_INTRODUCED_20,X_INTRODUCED_21,X_INTRODUCED_23];
array [1..22] of var int: X_INTRODUCED_54 ::var_is_introduced  = [X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_9,X_INTRODUCED_10,X_INTRODUCED_11,X_INTRODUCED_12,X_INTRODUCED_13,X_INTRODUCED_15,X_INTRODUCED_17,X_INTRODUCED_18,X_INTRODUCED_19,X_INTRODUCED_20,X_INTRODUCED_21,X_INTRODUCED_22,X_INTRODUCED_23,X_INTRODUCED_24];
array [1..26] of var int: X_INTRODUCED_63 ::var_is_introduced  = [X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_8,X_INTRODUCED_9,X_INTRODUCED_10,X_INTRODUCED_11,X_INTRODUCED_12,X_INTRODUCED_13,X_INTRODUCED_14,X_INTRODUCED_15,X_INTRODUCED_16,X_INTRODUCED_17,X_INTRODUCED_18,X_INTRODUCED_19,X_INTRODUCED_20,X_INTRODUCED_21,X_INTRODUCED_22,X_INTRODUCED_23,X_INTRODUCED_24,objective];
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_0,X_INTRODUCED_8],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_0,X_INTRODUCED_12],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_0,X_INTRODUCED_13],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_0,X_INTRODUCED_16],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_1,X_INTRODUCED_9],-2);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_1,X_INTRODUCED_20],-2);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_2,X_INTRODUCED_7],-2);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_2,X_INTRODUCED_20],-2);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_2,X_INTRODUCED_22],-2);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_3,X_INTRODUCED_6],-4);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_3,X_INTRODUCED_12],-4);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_3,X_INTRODUCED_19],-4);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_3,X_INTRODUCED_23],-4);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_4,X_INTRODUCED_9],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_4,X_INTRODUCED_11],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_4,X_INTRODUCED_15],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_4,X_INTRODUCED_16],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_4,X_INTRODUCED_19],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_4,X_INTRODUCED_21],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_5,X_INTRODUCED_9],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_5,X_INTRODUCED_12],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_5,X_INTRODUCED_23],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_6,X_INTRODUCED_11],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_6,X_INTRODUCED_20],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_6,X_INTRODUCED_21],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_7,X_INTRODUCED_10],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_7,X_INTRODUCED_13],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_8,X_INTRODUCED_9],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_8,X_INTRODUCED_17],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_10,X_INTRODUCED_13],-4);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_10,X_INTRODUCED_16],-4);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_10,X_INTRODUCED_18],-4);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_10,X_INTRODUCED_20],-4);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_11,X_INTRODUCED_14],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_12,X_INTRODUCED_17],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_12,X_INTRODUCED_22],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_13,X_INTRODUCED_16],-1);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_13,X_INTRODUCED_19],-1);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_15,X_INTRODUCED_17],-4);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_16,X_INTRODUCED_19],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_16,X_INTRODUCED_20],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_17,X_INTRODUCED_23],-1);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_18,X_INTRODUCED_22],-2);
constraint jacop_cumulative(X_INTRODUCED_48,X_INTRODUCED_47,X_INTRODUCED_46,5);
constraint jacop_cumulative(X_INTRODUCED_51,X_INTRODUCED_50,X_INTRODUCED_49,5);
constraint jacop_cumulative(X_INTRODUCED_54,X_INTRODUCED_53,X_INTRODUCED_52,6);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_9,objective],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_14,objective],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_19,objective],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_20,objective],-2);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_21,objective],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_22,objective],-3);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_23,objective],-5);
constraint int_lin_le(X_INTRODUCED_27,[X_INTRODUCED_24,objective],-4);
solve :: int_search(X_INTRODUCED_63,smallest,indomain_min,complete) minimize objective;
