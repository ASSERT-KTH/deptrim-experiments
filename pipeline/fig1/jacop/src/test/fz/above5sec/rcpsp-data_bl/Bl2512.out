objective = 31;
s = array1d(1..25, [0, 0, 2, 3, 4, 9, 17, 7, 3, 17, 12, 23, 14, 16, 26, 16, 17, 20, 22, 21, 29, 24, 26, 26, 19]);
----------

%% =====TIME-OUT=====
